tests/systemtests/mint_test.go

```diff
@@ -0,0 +1,97 @@
+package systemtests
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/sjson"
+)
+
+func TestMintQueries(t *testing.T) {
+	// scenario: test mint grpc queries
+	// given a running chain
+
+	sut.ResetChain(t)
+	cli := NewCLIWrapper(t, sut, verbose)
+
+	sut.ModifyGenesisJSON(t,
+		func(genesis []byte) []byte {
+			state, err := sjson.Set(string(genesis), "app_state.mint.minter.inflation", "1.00")
+			require.NoError(t, err)
+			return []byte(state)
+		},
+		func(genesis []byte) []byte {
+			state, err := sjson.Set(string(genesis), "app_state.mint.params.inflation_max", "1.00")
+			require.NoError(t, err)
+			return []byte(state)
+		},
+	)
+
+	sut.StartChain(t)
+
+	sut.AwaitNextBlock(t)
+
+	baseurl := sut.APIAddress()
+	blockHeightHeader := "x-cosmos-block-height"
+	queryAtHeight := "1"
+
+	// TODO: check why difference in values basequeryAtH on height between v1 and v2
+	if isV2() {
+		queryAtHeight = "2"
```

Comment:
lets open an issue about this so we dont forget it

Reply:
Opened https://github.com/cosmos/cosmos-sdk/issues/22302